src/game_battler.cpp

```diff
@@ -459,10 +459,11 @@ void Game_Battler::RemoveBattleStates() {
 	if (IsDead() &&
 		non_permanent(1)) {
 		ChangeHp(1);
```

Comment:
> 3.14: RM2000 + RM2003: After a battle is over, heal all the states that despite being "Persist after battle", still have more than 0% chance of recovery (turns and physical recovery don't matter)."

```cpp
ChangeHp(1)
states[0] = 0
```

With #1394 merged this will break (won't heal 1 HP anymore). Proposed solution: ``RemoveState(1)``